pkg/kubelet/apis/podresources/server.go

```diff
@@ -51,22 +51,25 @@ func NewPodResourcesServer(podsProvider PodsProvider, devicesProvider DevicesPro
 // List returns information about the resources assigned to pods on the node
 func (p *podResourcesServer) List(ctx context.Context, req *v1alpha1.ListPodResourcesRequest) (*v1alpha1.ListPodResourcesResponse, error) {
 	pods := p.podsProvider.GetPods()
-	podResources := make([]*v1alpha1.PodResources, len(pods))
+	podResources := make([]*v1alpha1.PodResources, 0, len(pods))
 
-	for i, pod := range pods {
+	for _, pod := range pods {
+		if (pod.Status.Phase != v1.PodRunning) && (pod.Status.Phase != v1.PodPending) {
```

Comment:
I think this is a stale pod status (if this is coming from the pod manager).

Reply:
Really? In my understanding, Kubelet Podmanager returns current status but Kubelet Containermanager doesn't.

Reply:
PodManager contains the status as observed through our watch of the pod object in the API Server.  It means there is a round trip required before it is reflected here.